### PR TITLE
fix: cant duplicate single select

### LINKED ISSUE #5195: 🐛 Bug: Single select duplicate column fails if default value is configured
### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Steps to reproduce ?

Single select duplicate column fails if default value is configured

### Desired Behavior

Column should be duplicated

### Project Details

Node: **v16.14.0**
Arch: **arm64**
Platform: **darwin**
Docker: **false**
RootDB: **pg**
PackageVersion: **0.105.3**

### Attachments

GET /api/v1/db/meta/views/vw_3lypildan0lifw/columns 200 2497 - 1.562 ms
columnAdd :: BadRequest [Error]: Default value ''c'' is not a select option.
    at Function.badRequest (/Users/rajuudava/NocoDB/nocodb/packages/nocodb/src/lib/meta/helpers/catchError.ts:41:11)
    at columnAdd (/Users/rajuudava/NocoDB/nocodb/packages/nocodb/src/lib/meta/api/columnApis.ts:476:25)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at /Users/rajuudava/NocoDB/nocodb/packages/nocodb/src/lib/meta/helpers/catchError.ts:6:14
POST /api/v1/db/meta/tables/md_030qj6avckfnma/columns 400 53 - 7.462 ms

### PR DESCRIPTION
**Pull request by sreehari2003**
## Change Summary

Provide summary of changes with issue number if any.
- closes #5195 

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Review by dstala — CHANGES_REQUESTED** [review]
after duplicate, cell shows 'default option value' on click
Screenshot 2023-09-01 at 8 10 48 PM

**Comment by o1lab** [comment]
@mertmit to review.

**Review by dstala — APPROVED** [review]

File: packages/nocodb/src/services/columns.service.ts [hunk 1]
```diff
@@ -1087,30 +1087,34 @@ export class ColumnsService {
...
                   );
                 }
               } else {
-                for (const cdf of colBody.cdf.split(',')) {
-                  if (!optionTitles.includes(cdf.replace(/'/g, "''"))) {
+                for (let cdf of colBody.cdf.split(',')) {
+                  // pg payload is coming as cdf = "'a'::text"
+                  if (driverType === 'pg') {
+                    cdf = cdf.split(':')[0];
+                  }
```
Comment (pranavxc) [inline packages/nocodb/src/services/columns.service.ts:1105]: I guess we have to do this before splitting ?

File: packages/nocodb/src/services/columns.service.ts [hunk 2]
```diff
@@ -1087,30 +1087,34 @@ export class ColumnsService {
...
               // handle single quote for default value
               if (driverType === 'mysql' || driverType === 'mysql2') {
                 colBody.cdf = colBody.cdf.replace(/'/g, "'");
               } else {
-                colBody.cdf = colBody.cdf.replace(/'/g, "''");
-              }
-
-              if (driverType === 'pg') {
-                colBody.cdf = `'${colBody.cdf}'`;
+                colBody.cdf = colBody.cdf.replace(/'/g, '');
```
Comment (pranavxc) [inline packages/nocodb/src/services/columns.service.ts:1117]: Check previous comment

**Review by pranavxc — CHANGES_REQUESTED** [review]

**Comment by DarkPhoenix2704** [comment]
#6323 Includes some fix for this.

**Comment by dstala** [comment]
> #6323 Includes some fix for this.

Revisit this PR after #6323 is merged

**Pull request closed without merging**